### ISSUE TITLE
getJSON helper would be nice.

### ISSUE THREAD

**Issue by stefanpenner** [instructure/ic-ajax#1]

**Comment by ryanflorence** [comment]
``` js
getJSON('/api/foo')
ajax('/api/foo')
```

When you use promises with `getJSON` and `ajax` they are identical, not sure what the benefit is.

**Comment by stefanpenner** [comment]
this ajax is somewhat lower level, as it fulfills with a pojo, which itself has a response object.

getJSON could be a higher level utility

``` js
  model: function(){ 
    return Ember.ajax('/api/foo'); // will setupController with the request pojo
  }
```

getJSON could be a higher level utility, which doesn't give u access to the request pojo.

``` js
  model: function(){ 
    return Ember.getJSON('/api/foo'); // will just work
  }
```

**Comment by ryanflorence** [comment]
Actually, I've been thinking of adding a setting like "resolveResponse" or something so you don't get the awkward object but just the response.

Edit: yes, what you just said, you beat me to the punch.

**Comment by stefanpenner** [comment]
another idea _may_ be `Ember._ajax` (or `Ember.http`) is the lower level, and `Ember.ajax` is the higher level?

**Comment by ryanflorence** [comment]
Something like: 

``` js
ajax(url, { responseOnly: true });
```

You'd want this for all methods, not just 'get'.

Maybe true is the default and then people that need access to the `jqXHR` can add the setting.

Or maybe ajax resolves the response by default, and then has a separate method for getting everything (instead of a setting).

``` js
ajax('foo/bar') // resolves the response only
ajax.raw('foo/bar') // resolves the object with response, textStatus, and jqXHR
```

I don't know.

edit: you beat me again, why are you not in IRC/IM? this is ridiculous.

**Comment by stefanpenner** [comment]
I like the raw idea. That way ajax will "just work" for what most people expect.

what should we reject with in the non-raw scenario? Another piece of information to add to this discussion is the promise proxy stuff (if used) does set its reason property: https://github.com/emberjs/ember.js/blob/master/packages/ember-runtime/lib/mixins/promise_proxy.js#L18

**Comment by ryanflorence** [comment]
closed by 928494bfaa9145fb520eebd8dfdea4086cc7ebec

**Comment by rwjblue** [comment]
Awesome!
